### PR TITLE
update-modules-snapshots

### PR DESCRIPTION
**Pull request by sateeshperi**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/methylseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by ewels — COMMENTED** [review]
Needs a changelog entry, especially for the updated modules.

File: tests/clipping/main.nf.test [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-                em_seq = true
-                clip_r1 = 2
-                outdir = "$outputDir"
-            }
-        }
-
-        then {
-            assert workflow.success
-            assert snapshot(workflow.trace.tasks().size()).match()
-            // TODO NFCORE_METHYLSEQ:METHYLSEQ:TRIMGALORE should include
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply tests/clipping/main.nf.test:19]: this test has been moved to `bismark.nf.test`

**Comment by Emiller88** [comment]
IMO, let's add the modules updates to the template update, and then rebase this off the template update branch again.

File: modules.json [hunk 2]
```diff
@@ -67,12 +67,12 @@
                     },
                     "methyldackel/extract": {
```
[the remark this answers is not in the document]
Comment (Emiller88) [reply modules.json:68]: I'm confused why the modules were changed/updated here, did they get updated since https://github.com/nf-core/methylseq/commit/4aa0e789cc6fd5c524fd356eb4d46d4555113607 ?

**Comment by sateeshperi** [comment]
agreed. it'll be much neater

**Comment by Emiller88** [comment]
It appears the software versions file isn't deterministic 🙃

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,9 +22,9 @@ jobs:
     if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/methylseq') }}"
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
         NXF_VER:
-          - "22.10.1"
```
Comment (Emiller88) [inline .github/workflows/ci.yml:27]: Same as fetchngs, we should lock down the version, otherwise we're going to be constantly bumping snapshots.

Oh, actually the software versions could be problematic, we might get too used to bumping things on a Nextflow version bump

File: bin/check_samplesheet.py
Comment (Emiller88) [inline bin/check_samplesheet.py:1]: Not sure where this is coming from

Comment (Emiller88) [inline bin/check_samplesheet.py:1]: Still not sure where this is coming from. 

It made me realize though, we need to change this to 2.8 now 😬

File: nf-test.config [hunk 4]
```diff
@@ -1,7 +1,6 @@
 config {
-
     testsDir "tests"
     workDir ".nf-test"
+    configFile "nextflow.config"
     profile "test,docker"
```
Comment (Emiller88) [inline nf-test.config:5]: We should probably pick a lane on this...

File: tests/bismark/main.nf.test [hunk 5]
```diff
@@ -49,12 +52,10 @@ nextflow_pipeline {
 
         then {
             assert workflow.success
-            assert snapshot(workflow.trace.tasks().size()).match()
-            // FIXME assert snapshot(workflow).match()
```
Comment (Emiller88) [inline tests/bismark/main.nf.test:53]: Should there be a test here?

File: tests/bismark/main.nf.test [hunk 6]
```diff
@@ -88,12 +96,10 @@ nextflow_pipeline {
...
```
Comment (Emiller88) [inline tests/bismark/main.nf.test:92]: Should this have a test? Or are we going to do these in a follow-up PR?

File: tests/clipping/main.nf.test
Comment (Emiller88) [inline tests/clipping/main.nf.test:1]: Why were these deleted?

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Looks good, a few questions and comments

File: tests/bismark/main.nf.test (hunk 5 above)
Comment (sateeshperi) [reply tests/bismark/main.nf.test:53]: this is the deleted portion

Comment (sateeshperi) [reply tests/bismark/main.nf.test:53]: or do you mean assert for trace.tasks().size() ?

File: tests/clipping/main.nf.test
Comment (sateeshperi) [reply tests/clipping/main.nf.test:1]: moved to `bismark/main.nf.test`
